File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        hm = {}
        for i in range(len(s)):
            if s[i] in hm:
                hm[s[i]][0] += 1
                hm[s[i]][1] = i 
            else:
                hm[s[i]] = [1, i]
    
        mn = float('inf')
        for key, value in hm.items():
            if value[0] == 1 and value[1] < mn:
                mn = value[1]

        return mn if mn != float('inf') else -1

File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.java
class Solution {
    public int firstUniqChar(String s) {
        int c = -1;
        for (int i=0; i<s.length(); i++) {
            int f = 0;
            for (int j=0; j<s.length(); j++) {
                if (i != j && s.charAt(i) == s.charAt(j)) {
                    f = 1;
                    break;
                }
            }
            if (f == 0) {
                c = i;
                break;
            }
        }

        return c;
    }
}